test/test_ops.py

```diff
@@ -514,12 +516,22 @@ def test_nms_cuda(self, iou, dtype=torch.float64):
             is_eq = torch.allclose(scores[r_cpu], scores[r_cuda.cpu()], rtol=tol, atol=tol)
         assert is_eq, err_msg.format(iou)
 
-    @needs_cuda
+    @pytest.mark.parametrize('device', cpu_and_gpu())
     @pytest.mark.parametrize("iou", (.2, .5, .8))
-    @pytest.mark.parametrize("dtype", (torch.float, torch.half))
-    def test_autocast(self, iou, dtype):
-        with torch.cuda.amp.autocast():
-            self.test_nms_cuda(iou=iou, dtype=dtype)
+    @pytest.mark.parametrize("dtype", (torch.bfloat16, torch.half))
+    def test_autocast(self, device, iou, dtype):
+        def test_nms_cpu(iou, dtype):
```

Comment:
Instead of creating this new one here, do you think we could just rely on `test_nms_ref` instead?
It doesn't accept a `dtype` parameter so we could add it if it's relevant, or alternatively we can just define `test_fn` as a partial function.

No strong opinion on this, we can also leave as is, even though it's a bit unfortunate that we have to define a new `test_nms_cpu` here

Reply:
Creating this new one here just to check whether the input is converted to float type. Is your suggestion to use `test_nms_ref` instread of `test_nms_cpu` here?  It's all ok for me. Thank you so much for your suggestion @NicolasHug

Reply:
>  test_nms_ref instread of test_nms_cpu here?

Yes :)

Reply:
Ok, I will modify it later.